modules/kubernetes/widget.go

```diff
@@ -0,0 +1,209 @@
+package kubernetes
+
+import (
+	"fmt"
+
+	"github.com/rivo/tview"
+	"github.com/wtfutil/wtf/utils"
+	"github.com/wtfutil/wtf/view"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Widget contains all the config for the widget
+type Widget struct {
+	view.TextWidget
+
+	objects    []string
+	title      string
+	kubeconfig string
+	namespaces []string
+	settings   *Settings
+}
+
+// NewWidget creates a new instance of the widget
+func NewWidget(app *tview.Application, settings *Settings) *Widget {
+	widget := Widget{
+		TextWidget: view.NewTextWidget(app, settings.common, false),
+
+		objects:    settings.objects,
+		title:      settings.title,
+		kubeconfig: settings.kubeconfig,
+		namespaces: settings.namespaces,
+		settings:   settings,
+	}
+
+	widget.View.SetWrap(true)
+
+	return &widget
+}
+
+// Refresh executes the command and updates the view with the results
+func (widget *Widget) Refresh() {
+
+	title := widget.generateTitle()
+
+	client := widget.getInstance()
+
+	var content string
+
+	// Debug Info
+	//content += fmt.Sprintf("Namespaces: %q %d\n", widget.namespaces, len(widget.namespaces))
+	//content += fmt.Sprintf("Objects: %q %d\n\n", widget.objects, len(widget.objects))
+
+	if !utils.DoesNotInclude(widget.objects, "nodes") {
```

Comment:
I really should add an `Includes()` function....

Reply:
I _almost_ wrote one instead of doing this, but I got lazy. :-D